streams/src/main/java/org/apache/kafka/streams/errors/ProductionExceptionHandler.java

```diff
@@ -56,10 +66,16 @@ default ProductionExceptionHandlerResponse handle(final ErrorHandlerContext cont
      * Handles serialization exception and determine if the process should continue. The default implementation is to
      * fail the process.
      *
-     * @param record        the record that failed to serialize
-     * @param exception     the exception that occurred during serialization
+     * @param record
+     *     The record that failed to serialize.
+     * @param exception
+     *     The exception that occurred during serialization.
+     *
+     * @return Whether to continue or stop processing, or retry the failed operation.
+     *
      * @deprecated Since 3.9. Use {@link #handleSerializationException(ErrorHandlerContext, ProducerRecord, Exception, SerializationExceptionOrigin)} instead.
      */
+    @SuppressWarnings({"rawtypes", "unused"})
```

Comment:
instead of using `rawtypes` it could be `ProducerRecord<?, ?>` but it's up to you

Reply:
We cannot... It's a know issue. (Would break backward compatibility -- it's a Java thing)